nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -535,6 +565,34 @@ nav_msgs::msg::Path RegulatedPurePursuitController::transformGlobalPlan(
   return transformed_plan;
 }
 
+double RegulatedPurePursuitController::findDirectionChange(
+  const geometry_msgs::msg::PoseStamped & pose )
+{
+  // Iterating through the global path to determine the position of the cusp
+   for (unsigned int pose_id = 1; pose_id <= global_plan_.poses.size(); ++pose_id)  {
```

Comment:
This should be `<` not `<=` since size() of 7 has elements of 0,1,2,3,4,5,6. I think in fixing that you can remove your most recent change of `|| pose_id == global_plan_.poses.size()`, yeah? Or if not, please explain why that's there. If this is actually the last point, then why bother detecting it?

Reply:
`|| pose_id == global_plan_.poses.size()`

This is very important. There maybe a scenario, where there is no cusp at all in the path (easiest example will be: imagine a robot driving straight to a goal). In that scenario, we need the distance of the goal from the robot's position, so that [here](https://github.com/padhupradheep/navigation2/blob/3c39b58b14e65642d2dd8040ec812d752e367dc5/nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp#L250), we don't satisfy this condition. Because the return value from this function where the code lies will be 0 and the robot will not move at all.

Reply:
Well the `<=` should definitely be `<`, if we can agree on that we can separate the topics 

---

I see. Would having the `lookahead_dist` be an argument to this function help? So instead of returning `0.0` if no cusp is found, instead it is `lookahead_dist`. That way if you iterate through the set and find no cusp, you exit the loop without hitting that condition and return the initial lookahead distance?

Reply:
> Well the <= should definitely be <, if we can agree on that we can separate the topics
Agreed, I think it makes sense. 

>  Honestly you bring up a good point, 0.0 is a bad return value. How about change return 0.0 to return std::numeric_limits<double>::max().

This is a valid point, which I would definitely change!